cdap-data-fabric/src/main/java/co/cask/cdap/data/format/SyslogRecordFormat.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.data.format;
+
+import co.cask.cdap.api.data.schema.Schema;
+import oi.thekraken.grok.api.Grok;
+
+import java.util.Map;
+
+/**
+ * GrokRecordFormat
+ */
+public class SyslogRecordFormat extends GrokRecordFormat {
+
+  @Override
+  protected Schema getDefaultSchema() {
+    return Schema.recordOf(
+      "streamEvent",
+      Schema.Field.of("timestamp", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("logsource", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("program", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("message", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("pid", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+  }
+
+  @Override
+  protected void addPatterns(Grok grok) {
+    addPattern(grok, "cdap/grok/patterns/grok-patterns");
+    addPattern(grok, "cdap/grok/patterns/linux-syslog");
```

Comment:
Seems like this duplication is unnecessary, especially this code is in the core data fabric module. Have you explore if there is any refactoring in the data fabric module to have it better supported for cases like this?

Sent from my iPhone

> On Mar 29, 2015, at 1:47 AM, Alvin Wang <notifications@github.com> wrote:
> 
> In cdap-data-fabric/src/main/java/co/cask/cdap/data/format/SyslogRecordFormat.java:
> 
> > +
> > +  @Override
> > +  protected Schema getDefaultSchema() {
> > +    return Schema.recordOf(
> > +      "streamEvent",
> > +      Schema.Field.of("timestamp", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
> > +      Schema.Field.of("logsource", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
> > +      Schema.Field.of("program", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
> > +      Schema.Field.of("message", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
> > +      Schema.Field.of("pid", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
> > +  }
> > +
> > +  @Override
> > +  protected void addPatterns(Grok grok) {
> > +    addPattern(grok, "cdap/grok/patterns/grok-patterns");
> > +    addPattern(grok, "cdap/grok/patterns/linux-syslog");
> It does, but this provides a simpler and quicker way to set stream format to syslog format since user doesn't have to specify the pattern (as there is currently no way to list all the patterns, so the user would have to read the source code).
> 
> —
> Reply to this email directly or view it on GitHub.
>